pre_commit_hooks/make/phony-targets.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Make environment variables working in OSX GUI apps such as Github Desktop
+# https://stackoverflow.com/q/135688/483528
+export PATH=$PATH:/usr/local/bin
+
+EXITCODE=0
+
+validate_phony_targets() {
+    local source=$1
+    local targets=$(sed -n 's#^.PHONY:\(.*\)#\1#p' <<< $source)
+
+    for target in $targets; do
+        # The q command for setting exit codes if a pattern doesn't match,
+        # isn't supported in OSX / BSD sed.
+        # We check the exact match as a workaround instead.
+        # If the pattern doesn't match that means that the target doesn't exist.
+        if test -z "$(sed -ne "s#^\(${target}:\).*#\1#p" <<< $source)"; then
+            echo "No target found for PHONY target: '$target'"
+            EXITCODE=1
+        fi
+    done
+}
+
+for file in "$@"; do
+    SOURCE="$(cat $file)"
```

Comment:
render make file to replace any variables and get the final phony targets and just capture all targets ;)

```suggestion
    SOURCE=$(make -npq -f "${file}" | grep -E $'^[^ \t#][^:#]*:')
```

this also captures complex makefile targets like:

```
aaa bbb ccc: $(shell ls Makefile*)
```

Reply:
Very good suggestion. However, we can't use it in this way. Make will actually always presume that a target that is marked as PHONY exists.

<img width="1435" alt="Screenshot 2020-05-25 at 14 19 43" src="https://user-images.githubusercontent.com/930643/82812503-6c258200-9e93-11ea-8b6f-d01f60f1014d.png">

Reply:
yeah then just check against the actual file.. ;) nice catch..

Reply:
resolved in 8176ec8

Reply:
this commit does not handle complex phony targets.. guess you closed the wrong conversation ;)